### PR TITLE
feat: add ushiftRight_*_distrib theorems

### PR DESCRIPTION
**Pull request by tobiasgrosser**

### CODE REVIEW

**Comment by semorrison** [comment]
This is still set as a draft PR. Is that intentional?

File: src/Init/Data/BitVec/Lemmas.lean [hunk 1]
```diff
@@ -700,6 +700,24 @@ theorem shiftLeft_shiftLeft {w : Nat} (x : BitVec w) (n m : Nat) :
     getLsb (x >>> i) j = getLsb x (i+j) := by
   unfold getLsb ; simp
 
+def ushiftRight_xor_distrib (a b c : BitVec w) :
```
[the remark this answers is not in the document]
Comment (tobiasgrosser) [reply src/Init/Data/BitVec/Lemmas.lean:703]: Fixed.

**Comment by tobiasgrosser** [comment]
> Mathlib CI status ([docs](https://leanprover-community.github.io/contribute/tags_and_branches.html)):
> 
> * ❗ Batteries/Mathlib CI will not be attempted unless your PR branches off the `nightly-with-mathlib` branch. Try `git rebase 14d59b3599d0b4fbfaf147e7ede2d82cbad7a1e7 --onto 647a5e94925791f6a16b629b6c16ffad60a7f478`. (2024-08-06 07:20:08)

I have now moved these theorems to work on Nat on the RHS. This seems to be the canonical form we are aiming for.

**Comment by tobiasgrosser** [comment]
awaiting-review

**Pull request merged**